src/mono/wasm/wasm.proj

```diff
@@ -125,10 +134,14 @@
 
     <ItemGroup>
       <_EmccLinkFlags Include="-s EXPORT_ES6=1" Condition="'$(WasmEnableES6)' == 'true'" />
+      <_EmccLinkFlags Include="-s LLD_REPORT_UNDEFINED" />
       <_EmccLinkFlags Include="-s ALLOW_MEMORY_GROWTH=1" />
       <_EmccLinkFlags Include="-s NO_EXIT_RUNTIME=1" />
       <_EmccLinkFlags Include="-s FORCE_FILESYSTEM=1" />
       <_EmccLinkFlags Include="-s EXPORTED_RUNTIME_METHODS=&quot;['FS','print','ccall','cwrap','setValue','getValue','UTF8ToString','UTF8ArrayToString','FS_createPath','FS_createDataFile','removeRunDependency','addRunDependency', 'FS_readFile']&quot;"  />
+      <_EmccCommonFlags Condition="'$(MonoWasmThreads)' == 'true'" Include="-s USE_PTHREADS=1" />
+      <_EmccLinkFlags Condition="'$(MonoWasmThreads)' == 'true'" Include="-Wno-pthreads-mem-growth" />
+      <_EmccLinkFlags Condition="'$(MonoWasmThreads)' == 'true'" Include="-s PTHREAD_POOL_SIZE=2" />
```

Comment:
We should probably make this configurable somehow.  Otherwise code like this will deadlock because the main thread never returns to the JS eventloop, so Emscripten can't start the third thread once it exhausts its initial pool:

```csharp
public static void Main() {
   Thread[] threads = new Thread[] { new Thread(ThreadMain1), new Thread(ThreadMain2) }
   threads[0].Start();
   threads[1].Start();
   threads[0].Join();
   threads[1].Join();
}
```

Reply:
The [docs](https://emscripten.org/docs/porting/pthreads.html#compiling-with-pthreads-enabled) indicate that `PTHREAD_POOL_SIZE` can be an integer or a javascript expression.  `navigator.hardwareConcurrency` might be a  better default.

Emscripten has some built in detection when the thread pool is exhausted.  `PTHREAD_POOL_SIZE_STRICT` by default warns when the thread pool is exhausted.  It can also be adjusted to error (`PTHREAD_POOL_SIZE_STRICT=2`) or do nothing (`PTHREAD_POOL_SIZE_STRICT=0`).

Reply:
Made an issue to revisit this https://github.com/dotnet/runtime/issues/68397